### PR TITLE
Feature/farm 67 switch profiles

### PR DESCRIPTION
**Pull request by isaacmartinezimv**
[[FARM-67 - Switch Profiles]](https://amidodevelopment.atlassian.net/secure/RapidBoard.jspa?rapidView=150&projectKey=FARM&view=planning&selectedIssue=FARM-67)

#### 📲 What

Screen to select switch what profile you want to load

#### 🤔 Why
		
Because we need the switch profile feature
		
#### 🛠 How
		
Using the WAMF Flutter code style

#### 👀 See
		
ezgif com-video-to-gif-2
		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [X] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

### CODE REVIEW

File: lib/ui/profile/SwitchProfile.dart [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        ),
+      ],
+    );
+  }
+
+  void _select(int index) {
+    setState(() {
+      widget._viewModel.selectedIndex = index;
+      _clearSelection();
+      widget._viewModel.actions[index].isSelected = true;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/SwitchProfile.dart:179]: the view shouldn't directly mutate the view model like this... this should be delegated to the viewmodel. viewModel.selectItem(itemViewmodel)

**Review by LeeMatthewHiggins — DISMISSED** [review]
We have some logic that should really be delegated to the viewModel. but lets get this in and we'll resolve when we integrate with the Bloc

File: lib/ui/profile/SwitchProfile.dart [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  static final EdgeInsets topButtonEdgePadding =
+      const EdgeInsets.only(right: 32.0);
+  static final EdgeInsets generalEdgePadding =
+      const EdgeInsets.only(left: 32, top: 10, bottom: 36);
+  static final EdgeInsets bottomButtonEdgePadding =
+      const EdgeInsets.only(right: 24, left: 24, bottom: 24);
+}
+
+class SwitchProfileViewModel {
+  String title;
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/SwitchProfile.dart:24]: these should all be final

**Review by davidfranquet — DISMISSED** [review]
Just lee's comment

File: lib/ui/profile/SwitchProfile.dart (hunk 2 above)
Comment (davidfranquet) [reply lib/ui/profile/SwitchProfile.dart:24]: yep

**Review by LeeMatthewHiggins — APPROVED** [review]
Lets get this in. But will need some changes when we integrate.

**Review by davidfranquet — APPROVED** [review]

**Pull request merged**